virttest/utils_misc.py

```diff
@@ -2739,31 +2738,49 @@ def format_windows_disk(session, did, mountpoint=None, size=None, fstype="ntfs")
     return False
 
 
-def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
+def format_linux_disk(session, did, mountpoint=None, size=None, fstype="ext3"):
     """
     Create a partition on disk in linux guest and format and mount it.
 
     :param session: session object to guest.
     :param did: disk serial, kname or wwn.
     :param mountpoint: mount point for the disk.
     :param fstype: filesystem type for the disk.
-    :param ostype: guest os type 'windows' or 'linux'.
     :return Boolean: disk usable or not.
     """
-    list_disk_cmd = "lsblk -p -oKNAME,MOUNTPOINT,SERIAL,SIZE,MODEL,WWN"
+    list_disk_cmd = "lsblk -oKNAME,MOUNTPOINT,SIZE,MODEL"
     disks = session.cmd_output(list_disk_cmd)
     logging.debug("Disks detail: %s" % disks)
-    for disk in disks.splitlines():
-        if did in disk:
-            kname = disk.split()[0]
-            logging.info("Create partition on disk '%s'" % kname)
-            mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
+    disks_splitlines = list(disks.splitlines())
+    disks_splitlines.pop(0)
+    for disk in disks_splitlines:
+        kname = disk.split()[0]
+        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
+        output = session.cmd_output_safe(get_disk_info_cmd)
+        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)| \
+                 (?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
+        did_list_group = re.finditer(re_str, output, re.M)
+        did_list = []
+        for i in did_list_group:
+            did_list.append(i.group())
+
+        if did in did_list:
```

Comment:
if did not in did_list:
     # search target disk
     continue
if not size:
     size = disk.split()[1] # better to replace disk with line